### PR TITLE
matched update_status_menu

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

File: src/80850_len_3060.c [hunk 1]
```diff
@@ -803,278 +803,300 @@ void update_status_menu(void) {
...
+    y = uiStatus->drawPosY + 28;
 
-    temp = uiStatus->displaySP / 256;
-    s4 = uiStatus->displaySP % 256;
-    s4 /= 32;
-    s4 = s4 + temp * 8;
+    spBars = uiStatus->displaySP / 256;
+    limit = uiStatus->displaySP % 256;
+    limit /= 32;
+    limit = limit + spBars * 8;
```
Comment (HailSanta) [inline src/80850_len_3060.c:856]: += here to be consistent with line 861

File: src/80850_len_3060.c [hunk 2]
```diff
@@ -803,278 +803,300 @@ void update_status_menu(void) {
...
     if (uiStatus->spBlinking) {
-        if (!s6) {
-            if (sp50 < uiStatus->spBarsToBlink * 8) {
+        if (!showStat) {
+            s32 limit = uiStatus->spBarsToBlink * 8;
+            do {} while (0);
+            if (sp50 < limit) {
                 while (TRUE) {
-                    s2++;
+                    i++;
```
Comment (HailSanta) [inline src/80850_len_3060.c:893]: I think I prefer the more concise formatting here, how does everyone else feel?

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.12%) on `us`.

File: src/world/partners.c [hunk 3]
```diff
@@ -153,14 +153,7 @@ HudScript* SPStarHudScripts[] = { &HES_StatusStar1, &HES_StatusStar3, &HES_Statu
...
 
-// TODO: make array when update_status_menu is decompiled
-s32 D_800F7FE8 = -1;
-s32 D_800F7FEC = 1;
-s32 D_800F7FF0 = 2;
-s32 D_800F7FF4 = 4;
-s32 D_800F7FF8 = 5;
-s32 D_800F7FFC = 7;
-s32 D_800F8000[] = { 8, 0, 0, 0 };
+s32 D_800F7FE8[] = { -1, 1, 2, 4, 5, 7, 8, 0, 0, 0 };
```
Comment (HailSanta) [inline src/world/partners.c:156]: Perhaps call this StatusMenuSPIncrementOffsets

**Review by HailSanta — APPROVED** [review]

File: src/80850_len_3060.c [hunk 4]
```diff
@@ -701,10 +683,10 @@ void update_status_menu(void) {
     }
 
     gDPSetScissor(gMainGfxPos++, G_SC_NON_INTERLACE, 12, 20, SCREEN_WIDTH - 12, SCREEN_HEIGHT - 20);
-    fp = uiStatus->drawPosX;
-    s5 = uiStatus->drawPosY;
-    draw_box(0, WINDOW_STYLE_5, fp,       s5, 0, 174, 35, 255, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, NULL, NULL, NULL, SCREEN_WIDTH, SCREEN_HEIGHT, NULL);
-    draw_box(0, WINDOW_STYLE_6, fp + 174, s5, 0, 122, 25, 255, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, NULL, NULL, NULL, SCREEN_WIDTH, SCREEN_HEIGHT, NULL);
+    x = uiStatus->drawPosX;
+    y = uiStatus->drawPosY;
+    draw_box(0, WINDOW_STYLE_5, x,       y, 0, 174, 35, 255, 0, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, NULL, NULL, NULL, 320, 240, NULL);
```
Comment (ethteck) [inline src/80850_len_3060.c:688]: SCREEN_WIDTH, SCREEN_HEIGHT like before

File: src/80850_len_3060.c (hunk 2 above)
Comment (ethteck) [reply src/80850_len_3060.c:893]: I think in this case it works, sure. but this is the kind of thing a code formatter would undo anyway, probably, so I don't think it matters much

**Review by ethteck — CHANGES_REQUESTED** [review]

**Pull request merged**